app/controllers/tutorials_controller.rb

```diff
@@ -1,6 +1,17 @@
 class TutorialsController < ApplicationController
+
   def show
     tutorial = Tutorial.find(params[:id])
-    @facade = TutorialFacade.new(tutorial, params[:video_id])
+    require_user! if tutorial.classroom
```

Comment:
Placing a check for current_user here is a great way to restrict the information displayed on the page.